protocol/messenger_messages.go

```diff
@@ -157,6 +157,21 @@ func (m *Messenger) DeleteMessageAndSend(ctx context.Context, messageID string)
 		return nil, ErrInvalidDeleteTypeAuthor
 	}
 
+	var messagesToDelete []*common.Message
```

Comment:
This piece of code that builds up the messages to be deleted is identical or very similar to what's in `messenger_handler.go` (I haven't checked thoroughly). I think we should at least extract it into a common function.